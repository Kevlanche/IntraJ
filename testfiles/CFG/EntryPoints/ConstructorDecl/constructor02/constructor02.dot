digraph G {
graph [splines=ortho, nodesep="1", ranksep="1"]
node [shape=rect, fontname=Arial];
{ rank = same; "Program[0]"}
{ rank = same; "List[1]"}
{ rank = same; "CompilationUnit[2]"}
{ rank = same; "List[3]"; "List[4]"}
{ rank = same; "ClassDecl[5]"}
{ rank = same; "Modifiers[6]"; "Opt[9]"; "List[10]"; "List[11]"}
{ rank = same; "List[7]"; "FieldDecl[12]"; "FieldDecl[21]"; "ConstructorDecl[30]"}
{ rank = same; "Modifier[8]"; "Modifiers[13]"; "PrimitiveTypeAccess[16]"; "List[17]"; "Modifiers[22]"; "PrimitiveTypeAccess[25]"; "List[26]"; "Entry[31]"; "Implicit[31]"; "Modifiers[31]"; "List[33]"; "List[42]"; "Opt[43]"; "Block[44]"; "Exit[31]"; "ExprStmt[59]"}
{ rank = same; "List[14]"; "FieldDeclarator[18]"; "List[23]"; "FieldDeclarator[27]"; "List[32]"; "ParameterDeclaration[34]"; "ParameterDeclaration[38]"; "List[45]"; "SuperConstructorAccess[60]"}
{ rank = same; "Modifier[15]"; "List[19]"; "Opt[20]"; "Modifier[24]"; "List[28]"; "Opt[29]"; "Modifiers[35]"; "PrimitiveTypeAccess[37]"; "Modifiers[39]"; "PrimitiveTypeAccess[41]"; "ExprStmt[46]"; "List[61]"}
{ rank = same; "List[36]"; "List[40]"; "Dot[47]"}
{ rank = same; "Dot[48]"; "MethodAccess[51]"}
{ rank = same; "TypeAccess[49]"; "VarAccess[50]"; "List[53]"; "List[57]"; "List[58]"; "UncheckedExceptions[52]"}
{ rank = same; "AddExpr[54]"}
{ rank = same; "VarAccess[55]"; "VarAccess[56]"}
"Program[0]" [label="Program\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[1]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"CompilationUnit[2]" [label="CompilationUnit\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[3]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[4]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"ClassDecl[5]" [label="ClassDecl\n < constructor02 >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifiers[6]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[7]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifier[8]" [label="Modifier\n < public >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Opt[9]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[10]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[11]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"FieldDecl[12]" [label="FieldDecl\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifiers[13]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[14]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifier[15]" [label="Modifier\n < private >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"PrimitiveTypeAccess[16]" [label="PrimitiveTypeAccess\n < int >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[17]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"FieldDeclarator[18]" [label="FieldDeclarator\n < i >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[19]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Opt[20]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"FieldDecl[21]" [label="FieldDecl\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifiers[22]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[23]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifier[24]" [label="Modifier\n < private >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"PrimitiveTypeAccess[25]" [label="PrimitiveTypeAccess\n < int >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[26]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"FieldDeclarator[27]" [label="FieldDeclarator\n < j >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[28]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Opt[29]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"ConstructorDecl[30]" [label="ConstructorDecl\n < constructor02 >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Entry[31]" [label="Entry\n ", fillcolor=white   style=filled  ]
"Implicit[31]" [label="Implicit\n < Instance >", fillcolor=white   style=filled  ]
"Modifiers[31]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[32]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[33]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"ParameterDeclaration[34]" [label="ParameterDeclaration\n < i >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifiers[35]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[36]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"PrimitiveTypeAccess[37]" [label="PrimitiveTypeAccess\n < int >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"ParameterDeclaration[38]" [label="ParameterDeclaration\n < j >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifiers[39]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[40]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"PrimitiveTypeAccess[41]" [label="PrimitiveTypeAccess\n < int >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[42]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Opt[43]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Block[44]" [label="Block\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[45]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"ExprStmt[46]" [label="ExprStmt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Dot[47]" [label="Dot\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Dot[48]" [label="Dot\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"TypeAccess[49]" [label="TypeAccess\n < System >", fillcolor=white   style=filled  ]
"VarAccess[50]" [label="VarAccess\n < out >", fillcolor=white   style=filled  ]
"MethodAccess[51]" [label="MethodAccess\n < println >", fillcolor=white   style=filled  ]
"List[53]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"AddExpr[54]" [label="AddExpr\n < + >", fillcolor=white   style=filled  ]
"VarAccess[55]" [label="VarAccess\n < i >", fillcolor=white   style=filled  ]
"VarAccess[56]" [label="VarAccess\n < j >", fillcolor=white   style=filled  ]
"List[57]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[58]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"UncheckedExceptions[52]" [label="UncheckedExceptions\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Exit[31]" [label="Exit\n ", fillcolor=white   style=filled  ]
"ExprStmt[59]" [label="ExprStmt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"SuperConstructorAccess[60]" [label="SuperConstructorAccess\n < super >", fillcolor=white   style=filled  ]
"List[61]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"CompilationUnit[2]" -> "List[3]" [style=dashed, color=gray]
"List[7]" -> "Modifier[8]" [style=dashed, color=gray]
"Modifiers[6]" -> "List[7]" [style=dashed, color=gray]
"ClassDecl[5]" -> "Modifiers[6]" [style=dashed, color=gray]
"ClassDecl[5]" -> "Opt[9]" [style=dashed, color=gray]
"ClassDecl[5]" -> "List[10]" [style=dashed, color=gray]
"List[14]" -> "Modifier[15]" [style=dashed, color=gray]
"Modifiers[13]" -> "List[14]" [style=dashed, color=gray]
"FieldDecl[12]" -> "Modifiers[13]" [style=dashed, color=gray]
"FieldDecl[12]" -> "PrimitiveTypeAccess[16]" [style=dashed, color=gray]
"FieldDeclarator[18]" -> "List[19]" [style=dashed, color=gray]
"FieldDeclarator[18]" -> "Opt[20]" [style=dashed, color=gray]
"List[17]" -> "FieldDeclarator[18]" [style=dashed, color=gray]
"FieldDecl[12]" -> "List[17]" [style=dashed, color=gray]
"List[11]" -> "FieldDecl[12]" [style=dashed, color=gray]
"List[23]" -> "Modifier[24]" [style=dashed, color=gray]
"Modifiers[22]" -> "List[23]" [style=dashed, color=gray]
"FieldDecl[21]" -> "Modifiers[22]" [style=dashed, color=gray]
"FieldDecl[21]" -> "PrimitiveTypeAccess[25]" [style=dashed, color=gray]
"FieldDeclarator[27]" -> "List[28]" [style=dashed, color=gray]
"FieldDeclarator[27]" -> "Opt[29]" [style=dashed, color=gray]
"List[26]" -> "FieldDeclarator[27]" [style=dashed, color=gray]
"FieldDecl[21]" -> "List[26]" [style=dashed, color=gray]
"List[11]" -> "FieldDecl[21]" [style=dashed, color=gray]
"ConstructorDecl[30]" -> "Entry[31]" [style=dashed, color=gray]
"ConstructorDecl[30]" -> "Implicit[31]" [style=dashed, color=gray]
"Modifiers[31]" -> "List[32]" [style=dashed, color=gray]
"ConstructorDecl[30]" -> "Modifiers[31]" [style=dashed, color=gray]
"Modifiers[35]" -> "List[36]" [style=dashed, color=gray]
"ParameterDeclaration[34]" -> "Modifiers[35]" [style=dashed, color=gray]
"ParameterDeclaration[34]" -> "PrimitiveTypeAccess[37]" [style=dashed, color=gray]
"List[33]" -> "ParameterDeclaration[34]" [style=dashed, color=gray]
"Modifiers[39]" -> "List[40]" [style=dashed, color=gray]
"ParameterDeclaration[38]" -> "Modifiers[39]" [style=dashed, color=gray]
"ParameterDeclaration[38]" -> "PrimitiveTypeAccess[41]" [style=dashed, color=gray]
"List[33]" -> "ParameterDeclaration[38]" [style=dashed, color=gray]
"ConstructorDecl[30]" -> "List[33]" [style=dashed, color=gray]
"ConstructorDecl[30]" -> "List[42]" [style=dashed, color=gray]
"ConstructorDecl[30]" -> "Opt[43]" [style=dashed, color=gray]
"Dot[48]" -> "TypeAccess[49]" [style=dashed, color=gray]
"Dot[48]" -> "VarAccess[50]" [style=dashed, color=gray]
"Dot[47]" -> "Dot[48]" [style=dashed, color=gray]
"AddExpr[54]" -> "VarAccess[55]" [style=dashed, color=gray]
"AddExpr[54]" -> "VarAccess[56]" [style=dashed, color=gray]
"List[53]" -> "AddExpr[54]" [style=dashed, color=gray]
"MethodAccess[51]" -> "List[53]" [style=dashed, color=gray]
"MethodAccess[51]" -> "List[57]" [style=dashed, color=gray]
"MethodAccess[51]" -> "List[58]" [style=dashed, color=gray]
"MethodAccess[51]" -> "UncheckedExceptions[52]" [style=dashed, color=gray]
"Dot[47]" -> "MethodAccess[51]" [style=dashed, color=gray]
"ExprStmt[46]" -> "Dot[47]" [style=dashed, color=gray]
"List[45]" -> "ExprStmt[46]" [style=dashed, color=gray]
"Block[44]" -> "List[45]" [style=dashed, color=gray]
"ConstructorDecl[30]" -> "Block[44]" [style=dashed, color=gray]
"ConstructorDecl[30]" -> "Exit[31]" [style=dashed, color=gray]
"SuperConstructorAccess[60]" -> "List[61]" [style=dashed, color=gray]
"ExprStmt[59]" -> "SuperConstructorAccess[60]" [style=dashed, color=gray]
"ConstructorDecl[30]" -> "ExprStmt[59]" [style=dashed, color=gray]
"List[11]" -> "ConstructorDecl[30]" [style=dashed, color=gray]
"ClassDecl[5]" -> "List[11]" [style=dashed, color=gray]
"List[4]" -> "ClassDecl[5]" [style=dashed, color=gray]
"CompilationUnit[2]" -> "List[4]" [style=dashed, color=gray]
"List[1]" -> "CompilationUnit[2]" [style=dashed, color=gray]
"Program[0]" -> "List[1]" [style=dashed, color=gray]
"Exit[31]" -> "MethodAccess[51]" [color=red, constraint=false]
"MethodAccess[51]" -> "Exit[31]" [color=blue, constraint=false, xlabel=" "] 
"MethodAccess[51]" -> "AddExpr[54]" [color=red, constraint=false]
"AddExpr[54]" -> "MethodAccess[51]" [color=blue, constraint=false, xlabel=" "] 
"AddExpr[54]" -> "VarAccess[56]" [color=red, constraint=false]
"VarAccess[56]" -> "AddExpr[54]" [color=blue, constraint=false, xlabel=" "] 
"VarAccess[56]" -> "VarAccess[55]" [color=red, constraint=false]
"VarAccess[55]" -> "VarAccess[56]" [color=blue, constraint=false, xlabel=" "] 
"VarAccess[55]" -> "VarAccess[50]" [color=red, constraint=false]
"VarAccess[50]" -> "VarAccess[55]" [color=blue, constraint=false, xlabel=" "] 
"VarAccess[50]" -> "TypeAccess[49]" [color=red, constraint=false]
"TypeAccess[49]" -> "VarAccess[50]" [color=blue, constraint=false, xlabel=" "] 
"TypeAccess[49]" -> "Implicit[31]" [color=red, constraint=false]
"Implicit[31]" -> "TypeAccess[49]" [color=blue, constraint=false, xlabel=" "] 
"Implicit[31]" -> "SuperConstructorAccess[60]" [color=red, constraint=false]
"SuperConstructorAccess[60]" -> "Implicit[31]" [color=blue, constraint=false, xlabel=" "] 
"SuperConstructorAccess[60]" -> "Entry[31]" [color=red, constraint=false]
"Entry[31]" -> "SuperConstructorAccess[60]" [color=blue, constraint=false, xlabel=" "] 
"List[3]" -> "List[4]" [style=invis]
"Modifiers[6]" -> "Opt[9]" -> "List[10]" -> "List[11]" [style=invis]
"FieldDecl[12]" -> "FieldDecl[21]" -> "ConstructorDecl[30]" [style=invis]
"Modifiers[13]" -> "PrimitiveTypeAccess[16]" -> "List[17]" [style=invis]
"List[19]" -> "Opt[20]" [style=invis]
"Modifiers[22]" -> "PrimitiveTypeAccess[25]" -> "List[26]" [style=invis]
"List[28]" -> "Opt[29]" [style=invis]
"Entry[31]" -> "Modifiers[31]" -> "List[33]" -> "List[42]" -> "Opt[43]" -> "Block[44]" -> "Exit[31]" [style=invis]
"ParameterDeclaration[34]" -> "ParameterDeclaration[38]" [style=invis]
"Modifiers[35]" -> "PrimitiveTypeAccess[37]" [style=invis]
"Modifiers[39]" -> "PrimitiveTypeAccess[41]" [style=invis]
"Dot[48]" -> "MethodAccess[51]" [style=invis]
"TypeAccess[49]" -> "VarAccess[50]" [style=invis]
"VarAccess[55]" -> "VarAccess[56]" [style=invis]

subgraph cluster_legend {
{node [style=filled, fillcolor=1, colorscheme="pastel13"]
Nodes [label="Total Nr. Nodes
65", fillcolor=1]
Edges [label="Total Nr. Edges
82", fillcolor=1]
NodesCFG [label="CFG Nr. Nodes
10", fillcolor=3]
EdgesCFG [label="CFG Nr. Edges
9", fillcolor=3]
Nodes-> "Program[0]" -> NodesCFG  [style="invis"]
Edges -> "Program[0]" ->EdgesCFG [style="invis"]}
label = "Statistics of the framework: NEXTFramework"
style="solid"
ranksep=0.05
nodesep=0.01
labelloc = b
len=2
}}