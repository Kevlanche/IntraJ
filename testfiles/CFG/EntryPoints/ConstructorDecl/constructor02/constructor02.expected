NodeName: Entry[31] Successors: [SuperConstructorAccess[60]]
NodeName: SuperConstructorAccess[60] Successors: [Implicit[31]]
NodeName: Implicit[31] Successors: [TypeAccess[49]]
NodeName: TypeAccess[49] Successors: [VarAccess[50]]
NodeName: VarAccess[50] Successors: [VarAccess[55]]
NodeName: VarAccess[55] Successors: [VarAccess[56]]
NodeName: VarAccess[56] Successors: [AddExpr[54]]
NodeName: AddExpr[54] Successors: [MethodAccess[51]]
NodeName: MethodAccess[51] Successors: [Exit[31]]
NodeName: Exit[31] Successors: []

NodeName: Exit[31] Predecessor: [MethodAccess[51]]
NodeName: MethodAccess[51] Predecessor: [AddExpr[54]]
NodeName: AddExpr[54] Predecessor: [VarAccess[56]]
NodeName: VarAccess[56] Predecessor: [VarAccess[55]]
NodeName: VarAccess[55] Predecessor: [VarAccess[50]]
NodeName: VarAccess[50] Predecessor: [TypeAccess[49]]
NodeName: TypeAccess[49] Predecessor: [Implicit[31]]
NodeName: Implicit[31] Predecessor: [SuperConstructorAccess[60]]
NodeName: SuperConstructorAccess[60] Predecessor: [Entry[31]]
NodeName: Entry[31] Predecessor: []
